src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -1913,17 +1930,21 @@ await _organizationUserRepository.DeleteManyAsync(removeUsersSet
 
             var newGroups = groups
                 .Where(g => !existingExternalGroupsDict.ContainsKey(g.Group.ExternalId))
-                .Select(g => g.Group);
+                .Select(g => g.Group).ToList();
 
+            var savedGroups = new List<Group>();
             foreach (var group in newGroups)
             {
                 group.CreationDate = group.RevisionDate = DateTime.UtcNow;
 
-                await _groupRepository.CreateAsync(group);
+                savedGroups.Add(await _groupRepository.CreateAsync(group));
                 await UpdateUsersAsync(group, groupsDict[group.ExternalId].ExternalUserIds,
                     existingExternalUsersIdDict);
             }
 
+            await _eventService.LogGroupEventsAsync(
+                savedGroups.Select(g => (g, EventType.Group_Created, eventSystemUser, (DateTime?)DateTime.UtcNow)));
+
```

Comment:
You also need to log the `Group_Updated` event for the next code block.

Reply:
Implemented on https://github.com/bitwarden/server/pull/4632/commits/e7d1f7eb2cff926994f61b91c88e553fe1bc8b6f